mParticle-Apple-SDK/Utils/NSDictionary+MPCaseInsensitive.swift

```diff
@@ -0,0 +1,141 @@
+//
+//  NSDictionary+MPCaseInsensitive.swift
+//  mParticle-Apple-SDK
+//
+//  Created by Brandon Stalnaker on 1/29/25.
+//
+
+import Foundation
+public extension NSDictionary {
+
+    @objc func caseInsensitiveKey(_ key: String) -> String? {
+        var localKey = key
+        
+        self.allKeys.forEach { keyValue in
+            if let stringKey = keyValue as? String, stringKey.caseInsensitiveCompare(key) == .orderedSame {
+                localKey = stringKey
+            }
+        }
+        return localKey
+    }
+
+    @objc func value(forCaseInsensitiveKey key: String) -> Any? {
+        var value : Any?
+        
+        self.allKeys.forEach { keyValue in
+            if let stringKey = keyValue as? String, stringKey.caseInsensitiveCompare(key) == .orderedSame {
+                value = self[stringKey]
+            }
+        }
+        
+        return value
+    }
+
+    @objc func transformValuesToString() -> [String : Any] {
+        let transformedDictionary: [String : Any] = self.reduce(into: [:]) { result, element in
```

Comment:
Double check that the use of the reduce method here doesn't bridge the NSDictionary to a Swift dict and mess up some of the types... The added unit test should catch that though.